### DISCUSSION TITLE
best way to learn programing

### DISCUSSION THREAD

**Discussion by Aiden-Tuite** [codewars/codewars.com#2446] [Q&A]
anyone have any free ways to learn im stuck and dont know anything

**Comment by Lockeheede** [comment]
Hi. Sorry you are stuck! I have been stuck myself for about 3 years and I'm finally starting to make things click. My trick is to read instead of doing video tutorials. The reason I'm saying this is because it seems harder to retain the information by simply following along videos. HOWEVER, once you have read a bit, do some videos to see if you can keep up with what they are talking about. If you want a free way, look up a book on coding in the library. Start with something short and "easy" (compared to C++ or Java), like JavaScript or Python. Maybe something project based would be the best way to start.  

Follow along in the book and whenever you get to a subject, like variables or functions, do their example BUT THEN, make your own examples. Keep making your own versions of the subjects until it starts to make sense. Study and practice your examples for 2-3 hours a day for a couple of weeks. 

Once you get to at least functions in your book, do some katas on codewars. Try to see how many 8 kyus you do until you are stumped. You just might rank up to 7 doing this (I just did that today! From 17 to 45 Honor!). After you rank up or get stumped, continue using your book. 

I can't guaruntee anything because I just starting figuring things out, but I can see I've gotten much better in a few days then I have in 3 years.
